### PR TITLE
Create isStagingDeployLocked action

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Creates an action to check if the StagingDeployCash is locked.

### Fixed Issues
Fixes (partial) https://github.com/Expensify/Expensify/issues/155140

### Tests
- [x] Run the test workflow with no open `StagingDeplyCash`. It should output "false" by default. ✅ https://github.com/Expensify/Expensify.cash/pull/1683/checks?check_run_id=2081644255
- [x] Open a test issue with the `StagingDeployCash` label in Expensify.cash ✅ https://github.com/Expensify/Expensify.cash/issues/1712
- [x] Re-run the test workflow from this PR. It should output "false". ✅ 
- [x] Add the `🔐 LockCashDeploys 🔐` label on the test `StagingDeployCash` ✅ 
- [x] Re-run the test workflow on this PR. It should output "true". ✅ https://github.com/Expensify/Expensify.cash/pull/1683/checks?check_run_id=2082052436

### Tested On
Github

### CODE REVIEW

**Comment by roryabraham** [comment]
Okay, this isn't working because I (foolishly!) forgot that we can't use ES6 modules in a Node.js script using the CJS module system. I _knew_ that I was going to forget and this was going to happen...

Anyways, I'm going to put together a PR to port over the ES6 `GithubUtils` and tests from `expensify-common` to `Expensify.cash/.github/libs/` and convert it to use CJS modules.

**Comment by roryabraham** [comment]
This needs to be HELD on https://github.com/Expensify/Expensify.cash/pull/1690

**Comment by roryabraham** [comment]
Oh, interesting. Now it's not working because the `OSBotify` user doesn't have read access to the `Expensify/Expensify` repo

**Comment by roryabraham** [comment]
Okay, this is now HELD on https://github.com/Expensify/Expensify/issues/157003

**Comment by roryabraham** [comment]
Okay, this is well-tested and ready for review

File: .github/actions/isStagingDeployLocked/isStagingDeployLocked.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const _ = require('underscore');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const GithubUtils = require('../../libs/GithubUtils');
+
+const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+const githubUtils = new GithubUtils(octokit);
+
+githubUtils.getStagingDeployCash()
+    .then(({labels}) => core.setOutput('IS_LOCKED', _.contains(_.pluck(labels, 'name'), '🔐 LockCashDeploys 🔐')))
```
Comment (AndrewGable) [inline .github/actions/isStagingDeployLocked/isStagingDeployLocked.js:10]: Not totally required to switch, but why make this it's own action when we could use something already built like: https://github.com/marketplace/actions/github-api-request ?

Comment (AndrewGable) [reply .github/actions/isStagingDeployLocked/isStagingDeployLocked.js:10]: Ah is it because this needs `getStagingDeployCash()`?

**Comment by Julesssss** [comment]
Tests seem good. 

But I'm confused by `isStagingDeployLocked/index.js`. Why do we need this full compiled code included here in the action? Is this simply the only way to depend on GithubAction functions? Would the alternative to publish the class as a library?

**Comment by roryabraham** [comment]
@Julesssss from the GH Actions docs:

> GitHub downloads each action run in a workflow during runtime and executes it as a complete package of code before you can use workflow commands like run to interact with the runner machine. This means you must include any package dependencies required to run the JavaScript code. You'll need to check in the toolkit core and github packages to your action's repository.

So we had a few options:

1) Commit `node_modules` ❌ 
2) Put each action in a separate repo, and commit the node modules in each repo.
3) Use `ncc` to precompile/bundle each action with its dependencies.

We've opted for (3)

File: .github/actions/isStagingDeployLocked/isStagingDeployLocked.js (hunk 1 above)
Comment (roryabraham) [reply .github/actions/isStagingDeployLocked/isStagingDeployLocked.js:10]: Yeah, it's just to take advantage of the code we've already built for fetching the open StagingDeployCash

**Comment by AndrewGable** [comment]
FYI conflicts

**Comment by roryabraham** [comment]
Merge conflicts resolved!

**Comment by Julesssss** [comment]
> We've opted for (3)

Thanks for explaining! Tough choice, no option is great 😞

**Review by Julesssss — APPROVED** [review]

**Comment by Julesssss** [comment]
LGTM. Will let Andrew approve and merge.

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**